Client/Frontend/Notifications/NotificationRootViewController.swift

```diff
@@ -159,9 +161,34 @@ private extension NotificationRootViewController {
         }
     }
 
-    @objc func didFinishSyncing() {
-        showingNotification = false
-        dispatch_async(dispatch_get_main_queue()) {
+    @objc func didFinishSyncing(notification: NSNotification) {
+        let syncMessage: NSAttributedString?
+        if let syncManager = notification.object as? SyncManager,
+            let syncDisplayState = syncManager.syncDisplayState {
+            switch syncDisplayState {
+            case .Bad(let message):
+                syncMessage = NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowErrorTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+            case .Stale(let message):
+                syncMessage = NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowWarningTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+            default: syncMessage = nil
+            }
+        } else {
+            syncMessage = nil
+        }
+
+        guard let displayMessage = syncMessage else {
+            showingNotification = false
+            return dispatch_async(dispatch_get_main_queue()) {
+                self.hideStatusNotification()
+            }
+        }
+        
+        notificationView.titleLabel.attributedText = displayMessage
+        notificationView.ellipsisLabel.hidden = true
+        self.notificationView.endAnimation()
+        let delayTime = dispatch_time(DISPATCH_TIME_NOW, Int64(4 * Double(NSEC_PER_SEC)))
+        dispatch_after(delayTime, dispatch_get_main_queue()) {
+            self.showingNotification = false
```

Comment:
Couple of things about this method:

1. The message passing from the NSNotification makes a lot of sense after our talk about bubbling up the error from the syncManager :+1:

2. I wonder if we can use the `defer` keyword in this method for the `dispatch_async` method instead of having it inline in the `guard` and at the end.

3. For the 4 second delay, I've done it a bit differently in my PR. Instead of using `dispatch_time`, I've added a `duration` parameter to the `showNotification` call that dismisses after 4 seconds. Internally it keeps a `NSTimer` on the common run loop this way if a sync finishes within the 4 seconds the notification is showing it dismisses it right away instead of waiting for the 4 seconds. Not sure this matters much though.

4. I'm a little wary passing in a reference to the `SyncManager` through the notification. If we're only using the `syncDisplayState`, could we just pass that instead?